pkg/handlers/default.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2020-2022 Kinvolk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handlers
+
+import (
+	"os"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/kinvolk/seccompagent/pkg/registry"
+	log "github.com/sirupsen/logrus"
+
+	libseccomp "github.com/seccomp/libseccomp-golang"
+)
+
+func KillContainer(pid int) registry.HandlerFunc {
+	return func(fd libseccomp.ScmpFd, req *libseccomp.ScmpNotifReq) (result registry.HandlerResult) {
+		p, err := os.FindProcess(pid)
+		if err != nil {
+			log.WithFields(log.Fields{
+				"pid": pid,
+			}).Error("cannot find process")
+			return registry.HandlerResultErrno(unix.ENOSYS)
+		}
```

Comment:
I think we should verify the seccomp notification is still valid here. So we at least reduce the race since the notification is valid (right now we know it was valid when it was received) and check for the pid. Otherwise, this pid might be from another process and we will do nasty things

Reply:
Ok. But note that `os.FindProcess` does not actually check the existence of the process on unix.